src/main/java/seedu/address/logic/commands/group/AddToGroupCommand.java

```diff
@@ -0,0 +1,131 @@
+            + "and will remain in the group, the rest of the users have been added accordingly: \n";
+    public static final String MESSAGE_GROUP_NOT_EXISTS = "There is no group with name %s.";
+
+    private final String groupName;
+
+    private final List<Index> members;
+
+    private List<Person> duplicates = new ArrayList<>();
+    private List<Person> newMembers = new ArrayList<>();
+
+    /**
+     * Creates a new AddToGroupCommand.
+     * @param groupName The group name
+     * @param members The persons to be included (referenced by index)
+     */
+    public AddToGroupCommand(String groupName, List<Index> members) {
+        this.groupName = groupName;
+        this.members = members;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getPersonList();
+        Group existingGroup;
+        try {
+            existingGroup = model.getGroup(groupName);
+        } catch (GroupNotFoundException gnfe) {
+            throw new CommandException(String.format(MESSAGE_GROUP_NOT_EXISTS, groupName));
+        }
+        Group newGroup = this.createGroupWithAddedMembers(existingGroup, members, lastShownList);
+        model.setGroup(existingGroup, newGroup);
+        try {
+            model.setGroup(existingGroup, newGroup);
+        } catch (DuplicateGroupException e) {
+            throw new CommandException(e.getMessage());
+        }
+        return new CommandResult(formatSuccessCommandResultMessage());
+    }
+
+    /**
+     * Formats the message to output to the user after the execution of the command.
+     * This is non-trivial as new entries and duplicates are taken into account to
+     * output the correct string format.
+     *
+     * @return a string which should go into the CommandResult.
+     */
+    private String formatSuccessCommandResultMessage() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(String.format(MESSAGE_ADD_TO_GROUP_SUCCESS, groupName));
+        this.newMembers.forEach(person -> builder.append(person.getName()).append('\n'));
+        if (!this.duplicates.isEmpty()) {
+            builder.append(String.format(MESSAGE_ADD_TO_GROUP_DUPLICATES, groupName));
+            this.duplicates.forEach(person -> builder.append(person.getName()).append('\n'));
+        }
+        return builder.toString();
+    }
+
+    /**
+     * To enforce immutability, this function creates a new group with the added members.
+     *
+     * @param existingGroup The group object of the group before addition of members.
+     * @param members The indices of the members to be added, based on the lastShownList.
+     * @param lastShownList The list of Person's objects last displayed to the user.
+     * @return a new Group object with the added person objects.
+     */
+    private Group createGroupWithAddedMembers(Group existingGroup,
```

Comment:
`createGroupWithAddedMembers` sounds a little strange to me. Maybe something like `createNewGroupWithMembers` or `createGroupWithExistingMembers`?

Reply:
I'm gonna name it createGroupWithNewMembers, because what it does is that it takes a group with all the old members, and it returns a group with the old members and the new members. Sounds the most intuitive to me.